substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_sun_security_ssl_TrustStoreManager.java

```diff
@@ -94,27 +109,95 @@ public void afterRegistration(AfterRegistrationAccess access) {
 @TargetClass(className = TrustStoreManagerFeature.TRUST_STORE_MANAGER_CLASS_NAME)
 final class Target_sun_security_ssl_TrustStoreManager {
 
+    @Alias @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.NewInstance, declClassName = TrustStoreManagerFeature.TRUST_STORE_MANAGER_CLASS_NAME +
+                    "$TrustAnchorManager") private static Target_sun_security_ssl_TrustStoreManager_TrustAnchorManager tam = null;
+
     @Substitute
     private static Set<X509Certificate> getTrustedCerts() throws Exception {
-        return ImageSingletons.lookup(TrustStoreManagerSupport.class).trustedCerts;
-    }
+        Set<X509Certificate> set = new HashSet<>(ImageSingletons.lookup(TrustStoreManagerSupport.class).trustedCerts);
```

Comment:
Well this could be debated, our use case is to include some corporate root CA at build time, because these are required to build and run in all our OpenShift clusters.
Then at run time we need to use some cluster specific certificates managed via cert manager to validate SSL services.

Reply:
In that case it was obvious for me to merge both sets... it also makes it easy to handle when `tam.getTrustedCerts` returns an empty one.